x-pack/platform/plugins/shared/cases/public/components/files/file_avatar.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { useFilesContext } from '@kbn/shared-ux-file-context';
+import { EuiAvatar } from '@elastic/eui';
+import type { FileJSON } from '@kbn/shared-ux-file-types';
+import type { Owner } from '../../../common/constants/types';
+import { constructFileKindIdByOwner } from '../../../common/files';
+import { useFilePreview } from './use_file_preview';
+import { FilePreview } from './file_preview';
+import { useCasesContext } from '../cases_context/use_cases_context';
+import { isImage } from './utils';
+
+// eslint-disable-next-line complexity
+const getIcon = (extension: string) => {
```

Comment:
@cesco-f I think it's hard to list all related extensions. We should instead use [parseMimeType](https://github.com/elastic/kibana/blob/1ea5d11cabf0876edded0f92703a470cb58930d6/x-pack/platform/plugins/shared/cases/public/components/files/utils.tsx#L24) function which is also used to display "Type" column in Files table

Reply:
The possible mime types are `imageMimeTypes`, `textMimeTypes`, `compressionMimeTypes` and `pdfMimeTypes`.

So I'm showing the `document` icon if the mime type is either `textMimeTypes` or `pdfMimeTypes`, the `folderClosed` icon if it's a `compressionMimeTypes`, the clickable image if it's `imageMimeTypes` and the `paperClip` icon in all the other cases.

Is this what you had in mind? Or how would you use the `parseMimeType` function you mentioned?